cmd/libs/go2idl/openapi-gen/generators/openapi.go

```diff
@@ -362,26 +362,68 @@ func (g openAPITypeWriter) generate(t *types.Type) error {
 	return nil
 }
 
-func (g openAPITypeWriter) generateExtensions(CommentLines []string) error {
+// getTagsMap extract the key-value pairs from openapi tags and struct tags returns a map[string]string
+func getTagsMap(CommentLines []string, tags string) (map[string]string, error) {
+	tagMap := make(map[string]string)
+
+	// Handle openapi tags
 	tagValues := getOpenAPITagValue(CommentLines)
-	anyExtension := false
 	for _, val := range tagValues {
 		if strings.HasPrefix(val, tagExtensionPrefix) {
-			if !anyExtension {
-				g.Do("VendorExtensible: spec.VendorExtensible{\nExtensions: spec.Extensions{\n", nil)
-				anyExtension = true
-			}
 			parts := strings.SplitN(val, ":", 2)
 			if len(parts) != 2 {
-				return fmt.Errorf("Invalid extension value: %v", val)
+				return nil, fmt.Errorf("Invalid extension value: %v", val)
 			}
-			g.Do("\"$.$\": ", parts[0])
-			g.Do("\"$.$\",\n", parts[1])
+			tagMap[parts[0]] = parts[1]
 		}
 	}
-	if anyExtension {
-		g.Do("},\n},\n", nil)
+
+	// Handle struct tags
+	tagPairs := strings.Fields(tags)
+	for _, tagPair := range tagPairs {
+		parts := strings.SplitN(tagPair, ":", 2)
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("Invalid extension value: %v", tagPair)
+		}
+		switch parts[0] {
+		case "patchStrategy":
```

Comment:
> Also instead of validating during openapi generation, why don't we just add a test using gengo. It shouldn't be hard to do.

My main concern with creating a separate validation for this is that it is yet another script that needs to be run as part of the code-generation phase.  The set of required scripts to run is already confusing and unclear, and adding another script worsens the problem.

Since the intention is for aggregated apiservers to use these tags, every aggregated apiserver project will need to run yet-another generation / validation script, whether they know it or not.

I'd prefer to make the open-api generation usable first and then making it generic once we have the framework in place to plugin customized validation phases.